src/tcp-intercept.cxx

```diff
@@ -473,6 +505,9 @@ int main(int argc, char* argv[]) {
 #if HAVE_DECL_IP_TRANSPARENT
 		int value = 1;
 		s_listen.setsockopt(SOL_IP, IP_TRANSPARENT, &value, sizeof(value));
+		s_listen.setsockopt(IPPROTO_TCP, TCP_NODELAY, (char *) &value, sizeof(value));
+		//Take care of socks hanging in Established state
+		s_listen.setsockopt(SOL_SOCKET,SO_KEEPALIVE, &value, sizeof(value));
```

Comment:
There is no need to set the listening socket with Keepalive. This socket will only receive new connections, so will never get out of the LISTENING state.

You should move this (and probably the NO_DELAY as well) to around here: https://github.com/maedot-d/tcp-intercept/blob/821a9f89a06c893c1648d547d2080e4b21a3e7ce/src/tcp-intercept.cxx#L271